plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,403 @@
+    }
+
+    /**
+     * support multiple document fields to map to the same model input fields,
+     * check if the key existed, then append to existed value to be a list.
+     */
+    private void updateModelParameters(String modelInputFieldName, String originalFieldValueAsString, Map<String, String> modelParameters) {
+
+        if (modelParameters.containsKey(modelInputFieldName)) {
+            Object existingValue = modelParameters.get(modelInputFieldName);
+            List<Object> updatedList = (List<Object>) existingValue;
+            updatedList.add(originalFieldValueAsString);
+            modelParameters.put(modelInputFieldName, toString(updatedList));
+        } else {
+            modelParameters.put(modelInputFieldName, originalFieldValueAsString);
+        }
+
+    }
+
+    private String getFieldPath(IngestDocument ingestDocument, String originalFieldName) {
+        final boolean fieldPathIsNullOrEmpty = Strings.isNullOrEmpty(originalFieldName);
+
+        if (fieldPathIsNullOrEmpty) {
+            return null;
+        }
+        final boolean hasFieldPath = ingestDocument.hasField(originalFieldName, true);
+        if (!hasFieldPath) {
+            return null;
+        }
+        return originalFieldName;
+    }
+
+    /**
+     * appendFieldValue to the ingestDocument without changing the source
+     */
+
+    private void appendFieldValue(
+        ModelTensorOutput modelTensorOutput,
+        String originalModelOutputFieldName,
+        String newDocumentFieldName,
+        IngestDocument ingestDocument
+    ) {
+        Object modelOutputValue = null;
+
+        if (modelTensorOutput.getMlModelOutputs() != null && modelTensorOutput.getMlModelOutputs().size() > 0) {
+
+            modelOutputValue = getModelOutputValue(modelTensorOutput, originalModelOutputFieldName, ignoreMissing);
+
+            List<String> dotPathsInArray = writeNewDotPathForNestedObject(ingestDocument.getSourceAndMetadata(), newDocumentFieldName);
+
+            if (dotPathsInArray.size() == 1) {
+                ValueSource ingestValue = ValueSource.wrap(modelOutputValue, scriptService);
+                TemplateScript.Factory ingestField = ConfigurationUtils
+                    .compileTemplate(TYPE, tag, newDocumentFieldName, newDocumentFieldName, scriptService);
+                ingestDocument.appendFieldValue(ingestField, ingestValue, false);
+            } else {
+                ArrayList<?> modelOutputValueArray;
+                if (modelOutputValue instanceof List) {
+                    List<?> modelOutputList = (List<?>) modelOutputValue;
+                    modelOutputValueArray = new ArrayList<>(modelOutputList);
+                } else if (modelOutputValue instanceof ArrayList) {
+                    modelOutputValueArray = (ArrayList<?>) modelOutputValue;
+                } else {
+                    throw new IllegalArgumentException("model output is not an array, cannot assign to array in documents.");
+                }
```

Comment:
simplify this part ?

```
if(!(modelOutputValue instanceof List)) {
    throw new IllegalArgumentException("Model output is not an array, cannot assign to array in documents.");
}

List<?> modelOutputValueArray = (List<?>) modelOutputValue;
```

Reply:
changed

Reply:
https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java#L330